Index: D:/lib/scripts/commit-email.pl
===================================================================
--- D:/lib/scripts/commit-email.pl	(revision 13)
+++ D:/lib/scripts/commit-email.pl	(revision 294)
@@ -39,6 +39,7 @@
 use strict;
 use Carp;
 use POSIX qw(strftime);
+use MIME::Base64;
 my ($sendmail, $smtp_server);
 
 ######################################################################
@@ -420,14 +421,16 @@
     ######################################################################
     # Assembly of log message.
 
-    if ($commondir ne '')
-      {
-        $subject_base = "r$rev - in $commondir: $dirlist";
-      }
-    else
-      {
-        $subject_base = "r$rev - $dirlist";
-      }
+    #if ($commondir ne '')
+    #  {
+    #    $subject_base = "r$rev - in $commondir: $dirlist";
+    #  }
+    #else
+    #  {
+    #    $subject_base = "r$rev - $dirlist";
+    #  }
+    $subject_base = "r$rev: $log[0]";
+    chomp($subject_base);	
 
     # Put together the body of the log message.
     push(@body, "Repository  : $repos\n");
@@ -557,8 +560,10 @@
     $formatted_date = strftime('%a, %e %b %Y %X %z', localtime());
     # push(@head, "Date: $formatted_date\n");
     push(@head, "To: $to\n");
-    push(@head, "From: $mail_from\n");
-    push(@head, "Subject: $subject\n");
+    push(@head, "From: $author <$mail_from>\n");
+    my $encoded_subject = encode_base64($subject);
+    $encoded_subject =~ s/\r?\n//g;
+    push(@head, "Subject: =?UTF-8?B?$encoded_subject?=\n");
     push(@head, "Reply-to: $reply_to\n") if $reply_to;
 
     ### Below, we set the content-type etc, but see these comments
